lib/src/native/realm_core.dart

```diff
@@ -3282,3 +3345,27 @@ extension PlatformEx on Platform {
     return result;
   }
 }
+
+/// @nodoc
+class SyncErrorDetails {
```

Comment:
Its not great that we are creating an instance just to invoke a create error method, but I see why you did it. lets leave it for now and we can refactor later when we deprecate the error ctors.
Besides this is on a slow code path, so it does not matter much.

Reply:
I added this class in order to structure better the parameters of function `createSyncError`. The other reason is that we have to extract the error details once from `realm_sync_error_code_t` and once from `realm_sync_error`. Having common class allows me to simplify the way how the parameters are passed.